src/sage/rings/function_field/drinfeld_modules/charzero_drinfeld_module.py

```diff
@@ -275,27 +291,36 @@ def logarithm(self, name='z'):
 
         INPUT:
 
+        - ``prec`` -- an integer or ``Infinity`` (default: ``Infinity``);
+          the precision at which the series is returned; if ``Infinity``,
+          a lazy power series in returned
+
         - ``name`` -- string (default: ``'z'``); the name of the
           generator of the lazy power series ring
 
-        OUTPUT: a lazy power series over the base field
-
         EXAMPLES::
 
             sage: A = GF(2)['T']
             sage: K.<T> = Frac(A)
             sage: phi = DrinfeldModule(A, [T, 1])
-            sage: log = phi.logarithm(); log
-            z + ((1/(T^2+T))*z^2) + ((1/(T^6+T^5+T^3+T^2))*z^4) + O(z^8)
 
-        The logarithm is returned as a lazy power series, meaning that
-        any of its coefficients can be computed on demands::
+        When ``prec`` is ``Infinity`` (which is the default),
+        the logarithm is returned as a lazy power series, meaning
+        that any of its coefficients can be computed on demands::
 
+            sage: log = phi.logarithm(); log
+            z + ((1/(T^2+T))*z^2) + ((1/(T^6+T^5+T^3+T^2))*z^4) + O(z^8)
             sage: log[2^4]
             1/(T^30 + T^29 + T^27 + ... + T^7 + T^5 + T^4)
             sage: log[2^5]
             1/(T^62 + T^61 + T^59 + ... + T^8 + T^6 + T^5)
 
+        On the contrary, when ``prec`` is a finite number, all the
```

Comment:
```suggestion
        If ``prec`` is a finite number, all the
```
(`on the contrary` would imply, to my non-native ear, that I assumed something which is not the case)